src/org/traccar/protocol/Tk103ProtocolDecoder.java

```diff
@@ -87,6 +94,20 @@ public Tk103ProtocolDecoder(Tk103Protocol protocol) {
             .any()
             .compile();
 
+    private static final Pattern PATTERN_LBSWIFI = new PatternBuilder()
+            .number("(d+),")                     // device id
+            .expression("(.{4}),")               // command
+            .number("(d+),")                     // mcc
+            .number("(d+),")                     // mnc
+            .number("(d+),")                     // lac
+            .number("(d+),")                     // cid
+            .number("(d+),")                     // number of wifi macs
+            .expression("((?:(?:[0-9A-Fa-f]{2}:){5}(?:[0-9A-Fa-f]{2})\\*[-+]?\\d{1,2}\\*\\d{1,2},)*)")
```

Comment:
This should be a number. Also please provide details on what this is. It seems like it can be split into separate items.

Reply:
According to documentation, there is a list of wifi MAC addresses of variable size. Any link to some samples how such list is defined in regex?

info from: https://goo.gl/8DKadX
```
1.6.1 Cell Tower & WIFI format: (DW50)
(013632651491,DW50,460,0,0,6,2,aa*bb*cc*dd*ee*ff:-
8:0,aa*bb*cc*dd*ee*ff:-8:0,040613,040137)
"(" stands for packet start
"013632651491" stands for tracker ID
"DW50" stands for data command
"460" mcc
"1" mnc
"0" lac
"6" cid
“2” wifi number
“aa*bb*cc*dd*ee*ff:-8:0” wifi string(mac:rssi:channel_number)
“aa*bb*cc*dd*ee*ff:-8:0” wifi string(mac:rssi:channel_number)
"040613" stands for data, 04/June/2013
"040137" stands for time, 04:01:37, UTC time
")" stands for packet end
```
in actual device the symbols "*" replaced by ":" and vice versa

Reply:
Well, at least it should be something like this:
```java
number("((?:(?:xx:){5}(?:xx)\\*[-+]?d+\\*d+,)*)");
```
or if you know it's always just MAC addresses:
```java
expression("(.*)");
```
and parse it in the code. This is what we do in some other decoders.

Reply:
ok, i will try to fix this...